src/client/dfuse/pil4dfs/dfs_dcache.c

```diff
@@ -0,0 +1,663 @@
+/**
+ * (C) Copyright 2022-2024 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#define D_LOGFAC DD_FAC(il)
+
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <daos/debug.h>
+#include <gurt/hash.h>
+#include <gurt/list.h>
+#include <daos_fs.h>
+#include <daos/common.h>
+
+#include "dfs_dcache.h"
+
+/** Size of the hash key prefix */
+#define DCACHE_KEY_PREF_SIZE 35
+#define DCACHE_KEY_MAX       (DCACHE_KEY_PREF_SIZE - 1 + PATH_MAX)
+
+#ifdef DAOS_BUILD_RELEASE
+
+#define DF_DK          "dk[%zi]"
+#define DP_DK(_dk)     strnlen((_dk), DCACHE_KEY_MAX)
```

Comment:
Is there a difference between `DK` and the `DE` macros that already exist?  `DE` was supposed to mean `directory entry` so is what I'd expect the cache to use.

Reply:
Ok, now I understand.  Thank you.